src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/Rules/CannotSealTypeTests.cs

```diff
@@ -0,0 +1,381 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ApiCompatibility.Abstractions;
+using Xunit;
+
+namespace Microsoft.DotNet.ApiCompatibility.Tests
+{
+    public class CannotSealTypeTests
+    {
+        [Theory]
+        [MemberData(nameof(SealNonInheritableTypeNotReportedData))]
+        public void SealNonInheritableTypeNotReported(string leftSyntax, string rightSyntax)
+        {
+            IAssemblySymbol left = SymbolFactory.GetAssemblyFromSyntax(leftSyntax);
+            IAssemblySymbol right = SymbolFactory.GetAssemblyFromSyntax(rightSyntax);
+
+            ApiComparer differ = new();
+            IEnumerable<CompatDifference> differences = differ.GetDifferences(left, right);
+
+            Assert.Empty(differences);
+        }
+
+        [Theory]
+        [MemberData(nameof(SealInheritableTypeReportedData))]
+        public void SealInheritableTypeReported(string leftSyntax, string rightSyntax)
+        {
+            IAssemblySymbol left = SymbolFactory.GetAssemblyFromSyntax(leftSyntax);
+            IAssemblySymbol right = SymbolFactory.GetAssemblyFromSyntax(rightSyntax);
+
+            ApiComparer differ = new();
+            IEnumerable<CompatDifference> differences = differ.GetDifferences(left, right);
+
+            CompatDifference[] expected = new[]
+            {
+                new CompatDifference(DiagnosticIds.CannotSealType, string.Empty, DifferenceType.Changed, "T:CompatTests.UnsealedClass")
+            };
+
+            Assert.Equal(expected, differences, CompatDifferenceComparer.Default);
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void SealInheritableTypeInternalsVisibleReported(bool includeInternals)
+        {
+            string leftSyntax = @"
+namespace CompatTests
+{
+    public class UnsealedClass
+    {
+        internal UnsealedClass() { }
+    }
+}
+";
+            string rightSyntax = @"
+namespace CompatTests
+{
+    public sealed class UnsealedClass
+    {
+      public UnsealedClass() { }
```

Comment:
Did you intend to make this public?  I don't think that's necessarily required for this test.